libs/s25main/buildings/nobMilitary.h

```diff
@@ -138,6 +140,10 @@ class nobMilitary : public nobBaseMilitary
     auto GetTroops() const { return helpers::nonNullPtrSpan(troops); }
     bool IsInTroops(const nofPassiveSoldier& soldier) const;
 
+    const unsigned* GetDesiredTroopsPointer(unsigned rank) const { return &desired_troops[rank]; }
```

Comment:
Not really a big fan of handing out a pointer. Maybe use `GetDesiredTroops` to update the UI texts periodically/before draw instead?

Reply:
This is for use with `Window::AddVarText`, I think this is how it is intended to be used currently. If `AddVarText` should be changed could that be a separate PR?